conversiontracerbullet/src/main/java/org/stanford/AuthDBProperties.java

```diff
@@ -19,22 +16,22 @@
 
     private static Logger log = LogManager.getLogger(AuthDBProperties.class.getName());
 
-    private static final String DEFAULT_CONFIG_FILE = "/server.conf";
+    private static final String PROPERTY_RESOURCE = "/server.conf";
 
     private String server = null;
     private String service = null;
     private String userName = null;
     private String userPass = null;
-    private String propertyFile = null;
+    private Properties properties = null;
```

Comment:
FYI, I went down that road before and ended up where it is now.  For example, if it gets passed, we get pseudo-code like this:
```
initProps( propertyFile ) {
  if propertyFile == null
    properties = loadResourceProperties
  else
    properties = loadPropertiesFile( propertyFile )
  ...
}
```
So, putting the initializers in the constructors avoids this null check.  I hope that helps?